db_stress_tool/db_stress_driver.cc

```diff
@@ -17,7 +17,8 @@ void ThreadBody(void* v) {
   ThreadState* thread = reinterpret_cast<ThreadState*>(v);
   SharedState* shared = thread->shared;
 
-  if (!FLAGS_skip_verifydb && shared->ShouldVerifyAtBeginning()) {
+  if (!FLAGS_skip_verifydb && shared->ShouldVerifyAtBeginning() &&
+      !FLAGS_post_verification_only) {
```

Comment:
Should we also apply this check to FLAGS_preserve_unverified_changes https://github.com/facebook/rocksdb/blob/main/db_stress_tool/db_stress_driver.cc#L62C1-L62C1 since FLAGS_preserve_unverified_changes works for pre-run verification only, which is skipped in FLAGS_post_verification_only = true?

Reply:
Done. I addressed this in the final commit.

Reply:
One future follow-up item to this PR: whether we should make verification-only run works with [FLAGS_preserve_unverified_changes](https://github.com/facebook/rocksdb/pull/10659) 

Upon verification failure in the verification-only run, it'd be good to retain the .state and .trace (files generated when unsynced data loss scenario is covered) from the timed-out run, especially if the time-out happens during its db operation phase or during its validation phase when the timed-out run has `preserve_unverified_changes=false` 
- If the time-out happens during its validation phase and the timed-out run has `preserve_unverified_changes=true`, we might as well reuse the left-behind \unverified folder for .state and .trace.

Making it work with FLAGS_preserve_unverified_changes is not that useful for db files IMO. This is because if the verification-only run fails, even without FLAGS_preserve_unverified_changes, the db files upon failure should be exactly the same files in the db we are interested in for debugging. However, the .state and .trace could have already been deleted in https://github.com/facebook/rocksdb/blob/main/db_stress_tool/expected_state.cc#L698-L707

Reply:
Seems like `Crash-recovery verification passed :)` will still be printed regardless of the result of the verification-only run.

```
Crash-recovery verification passed :)
Skipping database operations
2023/08/08-19:19:20  Starting verification
2023/08/08-19:19:20  Verification successful
```

By your [implementation](https://github.com/facebook/rocksdb/pull/11592/files#diff-018187ed6cad8991beb0d0fe1beb806b29e021e2c3303d2f4c24ad0ad4e1e276R21), the pre-run verification "Crash-recovery" in the msg refers to does not run at all . 

But the essence of the verification happened is indeed a crash-recovery (since (1) and (4) are the same as in https://github.com/facebook/rocksdb/pull/11592#discussion_r1284875081) . The confusing case would be we have `Crash-recovery verification passed :)` being printed while `Starting verification` actually fails.

This makes me wonder about the underlying code-design - is there any particular reason why the validation-only is implemented as a post-validation run instead of a pre-validation?